data-transfer-pipeline/extract.py

```diff
@@ -3,43 +3,41 @@
 from os import environ as ENV
 
 import pandas as pd
-import pyodbc
+import pymssql
 from dotenv import load_dotenv
 
 
 def create_connection():
     """Returns a connection to connect to the database. """
-    conn = pyodbc.connect(
-        'DRIVER={ODBC Driver 18 for SQL Server};'
-        f'SERVER={ENV['DB_HOST']};'
-        f'DATABASE={ENV['DB_NAME']};'
-        f'UID={ENV['DB_USER']};'
-        f'PWD={ENV['DB_PASSWORD']};'
-        'Encrypt=yes;'
-        'TrustServerCertificate=yes;'
-        # this workings in production but should be:
-        # f'TrustStore={path to pem certificate}' when deployed
-        # check how this would work in docker image
-    )
+    conn = pymssql.connect(
+        server=ENV['DB_HOST'],
+        database=ENV['DB_NAME'],
+        user=ENV['DB_USER'],
+        password=ENV['DB_PASSWORD'],
+        port=1433,
+        as_dict=True)
     return conn
 
 
 def extract_recordings() -> pd.DataFrame:
     '''Returns the recordings over 24 hours old as a dataframe.'''
     load_dotenv()
-
+    cutoff_time = datetime.now() - timedelta(days=1)
     query = f'''
     SELECT recording_id, plant_id, recording_taken, last_watered, soil_moisture, temperature
     FROM {ENV['SCHEMA_NAME']}.recording
-    WHERE recording_taken < ?
+    WHERE recording_taken < %s
     '''
 
-    cutoff_time = datetime.now() - timedelta(days=1)
     conn = create_connection()
+    with conn.cursor() as cursor:
+        cursor.execute(query, (cutoff_time,))
+        res = cursor.fetchall()
 
-    df = pd.read_sql(query, conn, params=[cutoff_time])
     conn.close()
+    df = pd.DataFrame(res)
```

Comment:
Could use a slightly nicer variable name for this

Reply:
damn the QA is QAing